### ISSUE TITLE
Invalid Sql Generation On query using both Principal and Primary keys

### ISSUE THREAD

**Issue by npbenjohnson** [dotnet/efcore#19817]
Minimal case included, the structure looks like:
Container -_1 to many via id key_-> Child -_Many to 1 via principal key_-> Parent -_1 to Many via primary key_-> Existential Strife

Sql Throws 'The multi-part identifier "p.Id" could not be bound.'

```C#
using Microsoft.EntityFrameworkCore;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System;
using System.Collections.Generic;
using System.Diagnostics.CodeAnalysis;
using System.Linq;
using System.Text;

namespace Test
{
    public class ChildList
    {
        public int Id { get; set; }
        public List<Child> Children { get; set; }
    }
    public class Parent
    {
        public int Id { get; set; }
        public string PrincipalKey { get; set; }
        public List<ExistentialStrife> ExistentialStrives { get; set; }
    }

    public class ExistentialStrife
    {
        public int Id { get; set; }
        public int ParentId { get; set; }
    }

    public class Child
    {
        public int Id { get; set; }
        public string PrincipalKey { get; set; }
        public int ChildContainerId { get; set; }
        public Parent Parent { get; set; }
    }

    public class Db : DbContext
    {
        public Db([NotNullAttribute] DbContextOptions options) : base(options)
        {
        }

        protected override void OnModelCreating(ModelBuilder modelBuilder)
        {
            modelBuilder.Entity<Child>().HasOne<Parent>().WithMany()
                .HasForeignKey(nameof(Child.PrincipalKey))
                .HasPrincipalKey(nameof(Parent.PrincipalKey));
            base.OnModelCreating(modelBuilder);
        }
        public DbSet<ChildList> ChildLists { get; set; }
        public DbSet<Parent> Parents { get; set; }
        public DbSet<Child> Children { get; set; }
    }
    [TestClass]
    public class EFTest
    {
        [TestMethod]
        public void EFSqlFailure()
        {
            const string connectionString = "Server=(localdb)\\mssqllocaldb;Database=TestDb;Trusted_Connection=True;MultipleActiveResultSets=true";

            var builder = new DbContextOptionsBuilder<Db>();
            builder.UseSqlServer(connectionString);
            var db = new Db(builder.Options);
            db.Database.EnsureDeleted();
            db.Database.EnsureCreated();

            var result = db.ChildLists.Include(c => c.Children).ThenInclude(c=>c.Parent).ThenInclude(p=>p.ExistentialStrives).FirstOrDefault();
        }
    }
}
```

### Further technical details

EF Core version: 3.1
Database provider: Microsoft.EntityFrameworkCore.SqlServer
Target framework: .NET Core 3.1
Operating system: Win 10
IDE: Visual Studio 2019 16.4

**Comment by ajcvickers** [comment]
@npbenjohnson I don't see any errors when I run your code with EF Core 3.1.1. The generated SQL is below. Did you reproduce this on 3.1.0 or 3.1.1?

```SQL
SELECT [t].[Id], [t0].[Id], [t0].[ChildContainerId], [t0].[ChildListId], [t0].[ParentId], [t0].[PrincipalKey], [t0].[Id0], [t0].[PrincipalKey0], [t0].[Id1], [t0].[ParentId0]
      FROM (
          SELECT TOP(1) [c].[Id]
          FROM [ChildLists] AS [c]
      ) AS [t]
      LEFT JOIN (
          SELECT [c0].[Id], [c0].[ChildContainerId], [c0].[ChildListId], [c0].[ParentId], [c0].[PrincipalKey], [p].[Id] AS [Id0], [p].[PrincipalKey] AS [PrincipalKey0], [e].[Id] AS [Id1], [e].[ParentId] AS [ParentId0]
          FROM [Children] AS [c0]
          LEFT JOIN [Parents] AS [p] ON [c0].[ParentId] = [p].[Id]
          LEFT JOIN [ExistentialStrife] AS [e] ON [p].[Id] = [e].[ParentId]
      ) AS [t0] ON [t].[Id] = [t0].[ChildListId]
      ORDER BY [t].[Id], [t0].[Id], [t0].[Id1]
```

**Comment by npbenjohnson** [comment]
I thought 3.1.1. but my SqlServer dll was still on 3.0 so this has already been resolved

**Comment by ajcvickers** [comment]
@smitpatel @maumar Is there a dupe for this that we fixed in 3.1.1?
